homeassistant/components/modbus/modbus.py

```diff
@@ -419,9 +419,15 @@ def pb_call(
         except ModbusException as exception_error:
             self._log_error(str(exception_error))
             return None
+        if not result:
+            self._log_error("Error: pymodbus returned None")
+            return None
```

Comment:
It would be nicer if you had a test that returned `None` instead of `ReadResult` to exercise this.

Reply:
I agree, will add that to my list of new tests, coming very soon.